extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/transaction/interceptor/MongoTransaction.java

```diff
@@ -0,0 +1,64 @@
+package io.quarkus.mongodb.panache.transaction.interceptor;
+
+import javax.transaction.HeuristicMixedException;
+import javax.transaction.HeuristicRollbackException;
+import javax.transaction.RollbackException;
+import javax.transaction.Status;
+import javax.transaction.Synchronization;
+import javax.transaction.SystemException;
+import javax.transaction.Transaction;
+import javax.transaction.xa.XAResource;
+
+import com.mongodb.client.ClientSession;
+
+public class MongoTransaction implements Transaction {
+    private ClientSession clientSession;
+    private int status = Status.STATUS_ACTIVE;
+
+    public MongoTransaction(ClientSession clientSession) {
+        this.clientSession = clientSession;
+    }
+
+    public ClientSession getClientSession() {
+        return clientSession;
+    }
+
+    @Override
+    public void commit() throws RollbackException, HeuristicMixedException, HeuristicRollbackException, SecurityException,
+            IllegalStateException, SystemException {
+        this.status = Status.STATUS_COMMITTING;
```

Comment:
Is it not up to this class to check the status and only commit if not already done, or not marked as rollback-only? Or is it the caller doing that?

Reply:
Hard to say, the Narayana code is not easy to follow as there is multiple layer of manager and tx classes.
I made several attemps before finding the way to do it, I will check again without this but as far as I remember without this the status didn't change.
Maybe someone that made the Narayana integration inside Quarkus can have a look ?

Reply:
@FroMage in fact there is no `setStatus` method inside the transaction object so there is no way for an external system to update the status of the transaction. This needs to be done from the inside of the transaction object.

Reply:
I checked on the Narayana and the Infinispan transaction object, and both do the same so it's fine.

Reply:
OK.